fv3config/config/diag_table.py

```diff
@@ -0,0 +1,212 @@
+import logging
+from typing import Sequence, Optional, Union
+import dataclasses
+import datetime
+import re
+
+import dacite
+
+from .._exceptions import ConfigError
+
+logger = logging.getLogger("fv3config")
+
+
+@dataclasses.dataclass
+class DiagTableField:
+    """Object representing a field of a diagnostics file."""
+
+    module_name: str
+    field_name: str
+    output_name: str
+    reduction_method: Optional[str] = "none"
+    time_sampling: Optional[str] = "all"
+    regional_section: Optional[str] = "none"
+    packing: Optional[int] = 2
+
+
+@dataclasses.dataclass
+class DiagTableFile:
+    """Object representing a diagnostics file."""
```

Comment:
"Object representing a diagnostics file" and the name DiagTableFile originally had me think this represented a file containing diagnostic data. Can this be renamed to fully convey what it is? I'd suggest DiagFileSpecification, DiagFileOutputSpec, or something else. If you use Spec or Specification I would suggest making a similar change/using the same word for the DiagTableField class.

It was hard for me to understand some of the documentation initially because of this confusion.

The Table part can be dropped, since on its own this class has nothing to do with a table.

Reply:
I got confused a bit about this naming as well. Over in fv3net, there is a distinction between `*Config` objects which the users configure directly, and the underlying objects which these configurations are constructed into.

Reply:
Instead of `Spec` I would opt for the term `Config` for consistency with the prognostic run runtime configurations.

Reply:
Good call, will add `Spec`.

I used `Table` because these classes are tightly coupled to the Fortran `diag_table`. E.g. I don't think we'd use these exact classes if we weren't trying to represent the `diag_table`.

Reply:
We might add a diagnostics manager for python diagnostics to fv3config, and it would be useful to have the `Table` distinction at that point.

Reply:
Looks like we cross streams. I'd still vote for "Config" , since `Spec` is a synonym and `config` is the existing convention (e.g this package is named`fv3config` not `fv3spec`).

Reply:
I agree.

Reply:
I'm good with `Config`.